.github/workflows/backup-cognito-users.yaml

```diff
@@ -0,0 +1,57 @@
+name: Create backup of cognito users
+on:
+  schedule: 
+    - cron: '0 1 * * 0'
+  workflow_dispatch:
+env:
+  region: 'eu-west-1'
+  backups-bucket: 'biomage-backups-production'
+  pool-name: 'biomage-user-pool-case-insensitive-production'
+
+jobs:
+  backup-users:
+    name: create backup of cognito users
+    runs-on: ubuntu-20.04
+    steps:
+      - id: setup-node
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+
+      - id: install
+        name: Install dependencies
+        run: |-
+          npm install -g cognito-backup-restore
+          sudo apt-get install jq
+
+      - id: set-up-creds
+        name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.region }}
+          
+      - id: backup-users
+        name: create backup json of cognito users
+        run: |-
+          userPoolId=$(aws cognito-idp list-user-pools --max-results 20 | jq -c '.UserPools[] | select(.Name | test("production")) | .Id')
+          cbr backup --region ${{env.region}} --key  ${{secrets.AWS_ACCESS_KEY_ID}} --secret ${{ secrets.AWS_SECRET_ACCESS_KEY }} --pool ${userPoolId} --directory .
+
+      - id: save-to-s3
+        name: save backup json to s3
+        run: |-
+          current_date=$(date '+%Y-%m-%d.%H-%M-%S')
+          userPoolId=$(aws cognito-idp list-user-pools --max-results 20 | jq -c '.UserPools[] | select(.Name | test(${{ env.pool-name }})) | .Id')
+          aws s3api put-object --bucket ${{env.backups-bucket}} --key ${current_date}/${{ env.pool-name }}/data.json --body ./${userPoolId}.json
```

Comment:
you don't need to use the pool ID here, you can just use the pool name and void having to call `list-user-pools`

Reply:
but the json file by the package created will have the userPoolId for a name